### PR TITLE
[Samples] Enhancements and cleanup

### PR DESCRIPTION
**Pull request by aserdean**
This PR takes care of a few tasks regarding the samples build system.

Fixes: https://github.com/Microsoft/openenclave/issues/1606
Fixes: https://github.com/Microsoft/openenclave/issues/1588
Fixes: https://github.com/Microsoft/openenclave/issues/1605
Fixes: https://github.com/Microsoft/openenclave/issues/1590
Fixes: https://github.com/Microsoft/openenclave/issues/1357

### CODE REVIEW

File: samples/Makefile [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-clean:
-	$(foreach i, $(DIRS), $(MAKE) -C $(i) clean $(NEWLINE) )
-
-run-silent:
-	@ $(foreach i, $(DIRS), $(MAKE) -C $(i) run $(NEWLINE) )
-
-run:
-	$(MAKE) -s run-silent
-
-world: clean build run
```
Comment (soccerGB) [inline samples/Makefile]: I am curious why this Makefile is being removed?

Comment (andschwa) [reply samples/Makefile]: It was only needed by the old script to run the builds, but now we run the builds via `test-samples.cmake` which invokes `make` for each sample's directory.

Comment (andschwa) [reply samples/Makefile]: This PR deletes `samples/test-samples.sh`.

Comment (andschwa) [reply samples/Makefile]: @soccerGB had a really good point about checking the validation script in oe-engine, but we're good, the script uses each sample's `Makefile`, it never used this one.

**Comment by aserdean** [comment]
Superseded-by: https://github.com/Microsoft/openenclave/pull/1638

**Comment by andschwa** [comment]
Thanks @aserdean, I figured you'd be asleep so I rebased for you, sorry!

**Comment by aserdean** [comment]
No worries, I'm glad this got in 😄 . Thanks for taking care of the rebase.

**Comment by andschwa** [comment]
@aserdean there were a few errors in the build unfortunately, I think we need to set PKG_CONFIG_PATH in the script. I might try this later today, but didn't get a chance to yet. The error:

```
Package oeenclave-clang++ was not found in the pkg-config search path.

Perhaps you should add the directory containing `oeenclave-clang++.pc'

to the PKG_CONFIG_PATH environment variable

No package 'oeenclave-clang++' found
```

Feel free to use my branch from the other PR, but I'll re-open this one for you too.

**Comment by andschwa** [comment]
Closing while I'm trying to fix my rebased copy.

**Pull request closed without merging**